packages/react/src/components/tag/Tag.tsx

```diff
@@ -22,114 +31,143 @@ export type TagProps = {
+   * Callback function fired when the tag is clicked. If set, a delete button will be shown.
    */
-  onDelete?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
-  /**
-   * Sets the role of the tag when it's clickable. Uses 'link' by default.
-   */
-  role?: 'link' | 'button';
+  onDelete?: (event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.KeyboardEvent<HTMLDivElement>) => void;
   /**
    * Size variant for the Tag.
-   * @default 'm'
+   * @default TagSize.Small
    */
-  size?: 'm' | 'l';
-  /**
-   * The label is only visible to screen readers. Can be used to give screen reader users additional information about the tag.
-   */
-  srOnlyLabel?: string;
+  size?: TagSize;
   /**
    * Custom theme styles
    */
   theme?: TagCustomTheme;
+  /**
+   * Ref to the root element
+   */
+  ref?: React.Ref<HTMLDivElement> | React.LegacyRef<HTMLAnchorElement>;
 };
 
-export const Tag = forwardRef<HTMLDivElement, TagProps>(
+export const Tag = forwardRef<HTMLDivElement | HTMLAnchorElement, React.ComponentProps<'div'> & TagProps>(
   (
     {
       children,
       className,
-      deleteButtonAriaLabel,
-      deleteButtonProps,
-      id = 'hds-tag', // Default value will be removed in the next major release
-      labelClassName,
-      labelProps,
       onClick,
       onDelete,
-      role = 'link',
-      size = 'm',
-      srOnlyLabel,
+      size = TagSize.Small,
       theme,
+      href,
+      iconLeft,
+      iconRight,
+      multiline = false,
       ...rest
     },
-    ref: React.Ref<HTMLDivElement>,
+    ref: React.Ref<HTMLDivElement | HTMLAnchorElement>,
   ) => {
+    let variant: TagVariant = 'informative';
+    if (typeof onClick === 'function' || typeof onDelete === 'function') {
+      variant = 'action';
+    } else if (href) {
+      variant = 'link';
+    }
+
     // custom theme class that is applied to the root element
     const customThemeClass = useTheme<TagCustomTheme>(styles.tag, theme);
     const largeClass = styles.large;
-    const containerClassName = classNames(styles.tag, size === 'l' && largeClass, customThemeClass, className);
-    const clickable = typeof onClick === 'function';
+    const containerClassName = classNames(
+      styles.tag,
+      size === TagSize.Large && largeClass,
+      customThemeClass,
+      className,
+      styles[variant],
+    );
+
+    const iconElementLeft = iconLeft ? (
+      <div className={classNames(styles.icon)} aria-hidden="true">
+        {iconLeft}
+      </div>
+    ) : null;
+
+    const iconElementRight = iconRight ? (
+      <div className={classNames(styles.icon)} aria-hidden="true">
+        {iconRight}
+      </div>
+    ) : null;
+
     const deletable = typeof onDelete === 'function';
-    const hideLabelFromScreenReaders = srOnlyLabel || deleteButtonAriaLabel;
+    const action = typeof onClick === 'function' || typeof onDelete === 'function';
```

Comment:
Better to prefix boolean consts with "is/has/can"
`const hasAction = !!(onClick ||  onDelete)`